dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/secret/ServiceIntegrationResourceTest.java

```diff
@@ -0,0 +1,253 @@
+        IntegrationTestInitService.getInstance().init();
+
+        final User user = mock(User.class);
+        when(user.getUserId()).thenReturn(ADMIN_DEFAULT_ID);
+        when(user.getEmailAddress()).thenReturn(ADMIN_DEFAULT_MAIL);
+        when(user.getFullName()).thenReturn(ADMIN_NAME);
+        when(user.getLocale()).thenReturn(Locale.getDefault());
+        when(user.isBackendUser()).thenReturn(true);
+
+        final WebResource webResource = mock(WebResource.class);
+        final InitDataObject dataObject = mock(InitDataObject.class);
+        when(dataObject.getUser()).thenReturn(user);
+
+        when(webResource.init(any(WebResource.InitBuilder.class))).thenReturn(dataObject);
+
+        ServiceIntegrationHelper serviceIntegrationHelper = new ServiceIntegrationHelper();
+
+        serviceIntegrationResource = new ServiceIntegrationResource(webResource, serviceIntegrationHelper);
+    }
+
+    private InputStream createServiceDescriptorFile(final String fileName, final String serviceKey,
+            final String serviceName, final String description, final boolean allowExtraParameters,
+            final Map<String, Param> paramMap) throws IOException {
+        final ServiceDescriptor serviceDescriptor = new ServiceDescriptor(serviceKey, serviceName, description, "/black.png", allowExtraParameters);
+
+        for (final Entry<String, Param> entry : paramMap.entrySet()) {
+            final Param param = entry.getValue();
+            serviceDescriptor.addParam(entry.getKey(),param.getValue(), param.isHidden(), param.getType(), param.getLabel(), param.getHint());
+        }
+
+        String basePath = System.getProperty("java.io.tmpdir");
+        basePath = Paths.get(basePath).normalize().toString();
+        final File file = new File(basePath,fileName);
+        ymlMapper.writeValue(file, serviceDescriptor);
+        //System.out.println(file);
+        return Files.newInputStream(Paths.get(file.getPath()));
+    }
+
+    private FormDataMultiPart createFormDataMultiPart(final String fileName, final InputStream inputStream) {
+        final FormDataBodyPart filePart1 = mock(FormDataBodyPart.class);
+        when(filePart1.getEntityAs(any(Class.class))).thenReturn(inputStream);
+        final ContentDisposition contentDisposition = mock(ContentDisposition.class);
+        when(contentDisposition.getFileName()).thenReturn(fileName);
+        when(filePart1.getContentDisposition()).thenReturn(contentDisposition);
+        final FormDataMultiPart formDataMultiPart = mock(FormDataMultiPart.class);
+        when(formDataMultiPart.getFields("file")).thenReturn(Collections.singletonList(filePart1));
+        return formDataMultiPart;
+    }
+
+    @Test
+    public void  Test_Create_service_descriptor_Then_Create_Service_Integration_Then_Delete_The_Whole_Integration() throws IOException {
+        final Map<String, Param> paramMap = ImmutableMap.of(
+                "p1", Param.newParam("v1", false, Type.STRING, "label", "hint"),
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The String literal "label" appears 6 times in this file; the first occurrence is on line 118](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4623512)